tests/CommandBusTest.php

```diff
@@ -67,4 +68,24 @@ public function testHandleThrowExceptionForInvalidCommand()
     {
         (new CommandBus([]))->handle('must be an object');
     }
+
+    public function testIfOneCanOnlyCreateWithValidMiddlewares()
+    {
+        $middlewareList = [$this->getMock('stdClass')];
+
+        try {
+            new CommandBus($middlewareList);
+            $this->fail('Should not be able to create a CommandBus without a valid list of middlewares!');
+        } catch (InvalidMiddlewareException $caught) {
```

Comment:
There is no need to use a try catch seehttps://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions also using a dataprovider to test for a string or null maybe better